spec/webvalve/manager_spec.rb

```diff
@@ -38,6 +38,9 @@
   end
 
   describe '#setup' do
+    let(:wildcard_substitution) { WebValve::ServiceUrlConverter::WILDCARD_SUBSTITUTION }
+    let(:url_suffix_pattern) { WebValve::ServiceUrlConverter::URL_SUFFIX_PATTERN }
```

Comment:
Worth flattening these into inlined test-only string patterns?

Reply:
Inlining it would also help the reader figure out what those lib constants actually resolve to. I had to pull up a REPL and play around with them to understand them.

Reply:
I'm not grokking your latest comment on this thread, can you go again? :)

Reply:
I think basically I like the quality of testing the url converter in integration with the manager, but I also like treating the Regexp code (and let's admit that it's code) as opaque to this test. I think what I want to do is stub the URL generator for most tests and have one integrated that does something like this but doesn't literalize the regexps unless we think they're highly stable? Maybe they are. I felt like I was being too creative with "what counts as a URL delimiter and what doesn't" to be confident of that stability when coding this, but maybe I actually have found a good definition of URL boundary char.

Reply:
Also, I think the test suite of the converter itself is the thing that should be going deep on what the heck, semantically, these Regexps do, right? I think there's good coverage of these behaviors in there and nobody reading this file should be investigating the Regexp behavior?

Reply:
> Also, I think the test suite of the converter itself is the thing that should be going deep on what the heck, semantically, these Regexps do, right? I think there's good coverage of these behaviors in there and nobody reading this file should be investigating the Regexp behavior?

agreed on this point for sure. ideally, this test suite could be refactored a bit into unit and integration tests where the unit tests for manager don't test anything that is covered by the behavior of the converter. and then there's an integration test for the manager or the library constant that tests the allow_url making assertions about the expected behavior in aggregate.

Reply:
Is this a nudge to do this test refactor now?

Reply:
e5b9568 is an attempt to segregate tests that benefit from integration vs mocking of the ServiceUrlConverter

Reply:
hah it wasn't really a nudge for that, no. just noting that it's a thing i should put on the todo list. 

but i will take a look at that commit

Reply:
I didn't do a super principled job at segregating the tests, I just looked at what failed in ways that weren't directly in the wheelhouse of my PR to fix when stubbed and threw them into the integration side. Happy to unwind it if you've got a grander vision there.